src/background/user-activity.ts

```diff
@@ -34,13 +34,27 @@ async function getIdleTimeout() {
   return preferences.autoLockTimeout;
 }
 
-export async function expireSessionIfNeeded() {
+export async function estimateSessionExpiry(): Promise<{
+  autoLockTimeout: number | 'none';
+  shouldExpire: boolean;
+  timeToExpiry: number;
+}> {
   const lastActive = await getLastActive();
   const autoLockTimeout = await getIdleTimeout();
   if (autoLockTimeout === 'none') {
-    return;
+    return { autoLockTimeout, shouldExpire: false, timeToExpiry: Infinity };
+  }
+  if (lastActive != null) {
+    const timeToExpiry = autoLockTimeout - (Date.now() - lastActive);
+    return { autoLockTimeout, shouldExpire: timeToExpiry < 0, timeToExpiry };
+  } else {
```

Comment:
We don't need `else` here, but this is a minor

Reply:
We can also combine `autoLockTimeout === 'none' && lastActive == null` in one condition